Framework/DataHandling/inc/MantidDataHandling/LoadSampleShape.h

```diff
@@ -43,14 +46,18 @@ class DLLExport LoadSampleShape : public Mantid::API::Algorithm {
   const std::string category() const override {
     return "DataHandling\\Instrument";
   }
+  boost::shared_ptr<Geometry::MeshObject>
+  rotate(boost::shared_ptr<Geometry::MeshObject> sampleMesh,
```

Comment:
It's good practice to try and avoid imposing the callers memory handling technique on functions and giving them more than they need. How about 
```cpp
void rotate(Geometry::MeshObject &sampleMesh, const Geometry::Goniometer & goniometer) const
```
?